core/state/iterator.go

```diff
@@ -27,7 +28,8 @@ import (
 )
 
 // nodeIterator is an iterator to traverse the entire state trie post-order,
-// including all of the contract code and contract state tries.
+// including all of the contract code and contract state tries. Preimage is
+// required in order to resolve the contract address.
```

Comment:
Are we sure this is safe? Preimages are not enabled by default AFAIK in Geth, so does this mean node iteration all of a sudden breaks with this PR?

Reply:
State iterator is useful to traverse the whole state, but it's not UX friendly, because the state keys are all hash instead of raw keys.  In order to provide some meaningful information, preimages are still required.

However, we already have the state dumper, IMO we should merge these two together and get rid of one of them.

Reply:
In case of this PR, state iterator is only used in test, to verify the state completeness, which I think it's a nice tool so I kept it.